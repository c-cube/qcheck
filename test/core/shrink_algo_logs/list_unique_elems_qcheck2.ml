open QCheck2

let list_unique_elems =
  Test.make ~name:"lists have unique elems"
    Gen.(list small_int)
    (Log.shrinks Print.(list int) @@
     fun xs -> let ys = List.sort_uniq Int.compare xs in
               List.length xs = List.length ys)
;;
try Test.check_exn list_unique_elems with Test.Test_fail _ -> ()
