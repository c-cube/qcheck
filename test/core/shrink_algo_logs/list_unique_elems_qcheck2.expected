fails [9; 3; 94; 3]
holds []
holds [7; 5]
holds [0; 8; 1]
fails [0; 3; 94; 3]
fails [0; 0; 94; 3]
fails [0; 0; 0; 3]
fails [0; 0; 0; 0]
